### ISSUE TITLE
Confusing about the two-stage fine-tuning process

### ISSUE THREAD

**Issue by TracyTd** [ZGC-LLM-Safety/TrafficLLM#10]
Dear authors,
Hello! I have a question regarding the two-stage fine-tuning process described in your work. Could you kindly help me understand how the two stages are connected during training? Specifically, I'm wondering whether it's necessary to add the --ptuning_checkpoint parameter to the second-stage training command.
Thank you very much for your time.

**Comment by CuiTianyu961030** [comment]
Thanks for your attention to our work!

In this work, the two-stage fine-tuning pipeline mainly aims to help LLM overcome multimodal learning between text and traffic data across different tasks. 

In our measurement, LLM suffers from learning multi-type semantics and traffic features at the same stage, leading to poor detection performance (10.2% average accuracy for three traffic detection tasks in our experiments).

In the first stage of the dual-stage tuning in TrafficLLM, we introduce natural language instruction tuning to inject the professional task description text from the field of cybersecurity into LLMs. In the second stage, we force TrafficLLM to learn the traffic pattern under the downstream tasks of traffic data. The two stages will build two types of PEFT models (NLP and downstream-task models) for inference.

After getting the PEFT models, TrafficLLM will use the NLP model to predict the downstream task name and call the corresponding task-specific models for downstream traffic pattern reasoning.

Maybe you will be confused about why the evaluation step only gives one `--ptuning_path` to input. In the evaluation, we only evaluate the accuracy of traffic detection across different downstream tasks instead of NLP capabilities. Therefore, you only need to configure the second-stage PEFT models in `--ptuning_path` for different downstream tasks.

**Comment by TracyTd** [comment]
Thank you very much!

**Comment by youbin2014** [comment]
Stage2 is expected to based on the stage1's ptuning file. However, in the current traning scripts of stage2, it didn't pass the stage1's peft model/path to the stage2's training, it seems like the trafficllm_stage2.sh should include the "--ptuning_checkpoint" argument so that stage2 is based on the stage1's peft to further fine-tune. 

like:

`
PRE_SEQ_LEN=128
LR=2e-2
NUM_GPUS=1
export CUDA_VISIBLE_DEVICES=1

torchrun --standalone --nnodes=1 --nproc-per-node=$NUM_GPUS main.py \
    --do_train \
    --train_file ../datasets/ustc-tfc-2016/ustc-tfc-2016_detection_packet_train.json \
    --validation_file ../datasets/ustc-tfc-2016/ustc-tfc-2016_detection_packet_train.json \
    --preprocessing_num_workers 10 \
    --prompt_column instruction \
    --response_column output \
    --overwrite_cache \
    --cache_dir ../cache \
    --model_name_or_path ../models/chatglm2/chatglm2-6b \
    --ptuning_checkpoint ./models/chatglm2/peft/instruction/checkpoint-8000\
    --output_dir ../models/chatglm2/peft/ustc-tfc-2016-detection-packet \
    --overwrite_output_dir \
    --max_source_length 1024 \
    --max_target_length 32 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 16 \
    --predict_with_generate \
    --max_steps 20000 \
    --logging_steps 10 \
    --save_steps 4000 \
    --learning_rate $LR \
    --pre_seq_len $PRE_SEQ_LEN
`


Let me know if I misunderstanding anything.
